docs/building-a-form/adding-widgets-and-definitions.md

```diff
@@ -0,0 +1,208 @@
+# Adding widgets and definitions
+
+There are many common fields and widgets you can use to build forms. There are common types of definitions: `schema`/`uiSchema` objects and functions that return `schema`/`uiSchema` objects. For the function versions, there is documentation within the fields for the parameters.
+
+### In this guide
+
+- [Common widgets](#common-widgets)
+- [Common definitions](#common-definitions)
+
+### Common widgets
+
+These common widgets are included in the schemaform by default. Set these widgets in a config file while building your form. Some are associated with particular schema types or formats. There are additional widgets available, but the schemaform uses definitions they're included in, rather than just the field.
+
+To use a widget, set `ui:widget` for a field to the name of the widget. Widgets are located in [/src/js/widgets](src/js/widgets).
+
+Widget       | Default schema type or format
```

Comment:
I copied these from /src/js/widgets, but need some help filling out the info in the second column. I'm also not positive that these are all intended to be listed.

I'm tempted to format these similarly to the definitions reference information below, mostly because they could possibly need descriptions. Reading the names is fairly self-explanatory, but I'm guessing there's some unique information about how these are implemented that might be useful to describe.

Reply:
Yes, I agree, there's more info we can capture here that would be useful. I think the second column should rather describe how to use each one. For the ones below where you've already listed something in the second column, those widgets are used by default when the schema looks a certain way (e.g., `CheckboxWidget` is used when the schema has `type: boolean`). For the others, they can either be set directly by doing `ui:widget` in the schema, or used through an existing definition (e.g, for `PhoneWidget`, it can be used by used the definition for phone, under `definitions/phone.js`.

I think we can fill out this table in a separate PR so it doesn't hold up this one :).

Reply:
Opened a new issue for this in https://github.com/usds/us-forms-system/issues/138. 👍